src/utils/propsAreEqual.ts

```diff
@@ -7,7 +7,25 @@ export function propsAreEqual<Props extends SketchProps = SketchProps>(
   previous: P5WrapperProps<Props>,
   next: P5WrapperProps<Props>
 ) {
-  const differences = diff(previous, next);
+  // set deepCompare to true if not specified by props
+  const deepCompare = next.deepCompare === undefined ? true : next.deepCompare;
+
+  // shallow
+  if (!deepCompare) {
+    for (const key in previous) {
+      if (previous[key] !== next[key]) {
+        return false;
+      }
+    }
+    for (const key in next) {
+      if (!(key in previous) && next[key] !== undefined) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
I would move these to their own function definition above of the `propsAreEqual` definition. Something like:

```ts
function shallowEqual<T>(previous: T, next: T) {
  const previousKeys = Object.keys(previous);
  const nextKeys = Object.keys(next);

  if (previousKeys.length !== nextKeys.length) {
    return false;
  }

  for (const key of previousKeys) {
     if (previous[key] !== next[key]) {
       return false;
     }
   }

  return true;
}

function deepEqual<T>(previous: T, next: T) {
  return diff(previous, next).length === 0;
}

export function propsAreEqual<Props extends SketchProps = SketchProps>(
   previous: P5WrapperProps<Props>,
   next: P5WrapperProps<Props>
) {
  const deepCompare = next.deepCompare ?? true;

  return deepCompare ? deepEqual(previous, next) : shallowEqual(previous, next);
}
```

Something like that but you can adapt it as you like. I just think the abstraction makes things a bit cleaner. Also note that I wrote this here and didn't actually test this code so please check and test and update if necessary.

Reply:
Hey, so sorry for being inactive on the matter for a while. Interestingly, I managed to find a work-around. Instead of passing down the synth object - i pass down a function, that does absolutely nothing, but return the synth object.. This seems to have resolved the issue and by using useCallback I can change the function when the synth address changes hence mimick a shallow comparison. Up to now that seems  to work fine and has not introduced any problems . It is however a not very elegant solution of course, neither very idiomatic, so if you want I can keep working on this PR to setup a cleaner way to bypass deep comparisons. Let me know what you think

Reply:
Yeah, keep exploring this PR if you're up for it, I'm curious what you come up with. I'm also curious to see your current implementation described above if you have a link. Indeed it is a solution also.

Reply:
the implementation of that solution is truly very simple, its just that:

<img width="436" alt="image" src="https://user-images.githubusercontent.com/103147851/225716258-c64ad0c8-b2d1-43de-a150-f82d086d94e3.png">

Reply:
Added a new PR and addressed your comments, let me know what you think :)